### PR TITLE
JAMES-2537 Introduce a mail queue configuration

### PR DESCRIPTION
**Pull request by aduprat**

### CODE REVIEW

**Comment by mbaechler** [comment]
And what about the management URI ?

File: dockerfiles/run/guice/cassandra-ldap/destination/conf/rabbitmq.properties [hunk 1]
```diff
@@ -0,0 +1 @@
+uri=
```
Comment (rouazana) [inline dockerfiles/run/guice/cassandra-ldap/destination/conf/rabbitmq.properties]: why do you need to add these meaningless files?

Comment (aduprat) [reply dockerfiles/run/guice/cassandra-ldap/destination/conf/rabbitmq.properties]: I added the uri has it should be (the binding in the docker will be done when RabbitMQ will be really needed).

**Comment by aduprat** [comment]
> And what about the management URI ?

Ok

**Comment by aduprat** [comment]
I remove the configuration from the _AmqpForwardAttribute_ mailet, has the management uri is not required in this case.

File: backends-common/rabbitmq/pom.xml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        <dependency>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>commons-configuration</groupId>
+            <artifactId>commons-configuration</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>commons-logging</groupId>
```
Comment (mbaechler) [inline backends-common/rabbitmq/pom.xml]: NOOOOOO !
Please, don't introduce this lib back, I fought so much to exclude it from every deps.
You should you the slf4j commons-logging bridge instead

File: backends-common/rabbitmq/src/main/java/org/apache/james/backend/mailqueue/RabbitMQConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    private static final String URI = "uri";
+    private static final String MANAGEMENT_URI = "management.uri";
+
+    public static RabbitMQConfiguration from(PropertiesConfiguration configuration) {
+        String uri = configuration.getString(URI);
+        Preconditions.checkState(!Strings.isNullOrEmpty(uri), "You need to specify the URI of RabbitMQ");
+
+        String managementUri = configuration.getString(MANAGEMENT_URI);
+        Preconditions.checkState(!Strings.isNullOrEmpty(managementUri), "You need to specify the management URI of RabbitMQ");
+        return new RabbitMQConfiguration(uri, managementUri);
```
Comment (mbaechler) [inline backends-common/rabbitmq/src/main/java/org/apache/james/backend/mailqueue/RabbitMQConfiguration.java]: what about checking they are actual URI by using URI object ?

Comment (aduprat) [reply backends-common/rabbitmq/src/main/java/org/apache/james/backend/mailqueue/RabbitMQConfiguration.java]: Ok, but I had to push -f in order to solve fixup conflicts.

**Comment by aduprat** [comment]
```
[ERROR]   CassandraMailboxRecentDAOTest.getRecentMessageUidsInMailboxShouldNotTimeoutWhenOverPagingLimit:142 
Expected size:<6000> but was:<7000> in:
```
test this please

File: core/pom.xml [hunk 4]
```diff
@@ -57,10 +57,18 @@
             <groupId>com.sun.mail</groupId>
             <artifactId>javax.mail</artifactId>
         </dependency>
+        <dependency>
+            <groupId>commons-configuration</groupId>
```
Comment (mbaechler) [inline core/pom.xml]: why do you need to touch this pom ?

File: core/pom.xml [hunk 5]
```diff
@@ -57,10 +57,18 @@
...
+            <artifactId>commons-configuration</artifactId>
+        </dependency>
         <dependency>
             <groupId>commons-io</groupId>
             <artifactId>commons-io</artifactId>
         </dependency>
+        <dependency>
+            <groupId>commons-logging</groupId>
```
Comment (mbaechler) [inline core/pom.xml]: and I'm still not happy with that

**Comment by chibenwa** [comment]
```
[ab9ff1b46cad6405a563ed4f2a843019465ca910] [ERROR] Errors: 
[ab9ff1b46cad6405a563ed4f2a843019465ca910] [ERROR]   SpamAssassinContract.spamShouldBeDeliveredInSpamMailboxOrInboxWhenMultipleRecipientsConfigurations:610 » ConditionTimeout
```

And after 02:10 h of build :trollface: 

test this please

**Pull request merged**